### ISSUE TITLE
Do use use Google DNS for IPv6 probes

### ISSUE THREAD

**Issue by rht** [brave/brave-browser#2324]
## Description 

Currently, the browser pings Google's IPv6 DNS servers to do probing. Note that, relatedly, ungoogled-chromium has a command line flag to disable the probing altogether, but changing the DNS server is another less intrusive solution.

## Steps to Reproduce

N/A

## Reproduces how often: 

Always

## Brave version (brave://version info)

HEAD

### Reproducible on current release:
- Does it reproduce on brave-browser dev/beta builds? 

### Website problems only:  
- Does the issue resolve itself when disabling Brave Shields? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by tomlowenthal** [comment]
Looks like this probe involves making a non-interactive DNS query to the k root name server — which is operated by RIPE NCC (not Google) — that's what this patch disables. First of all, do I have that right, or have I misunderstood? Would I be correct in understanding that this is a concerning behavior because it's a non-user-initiated DNS query.

**Comment by rht** [comment]
> First of all, do I have that right, or have I misunderstood?

Yeah, the patch does not disable the probing. I think I want to limit the scope of the issue on whether to not use Google's DNS server.

(Aside, I found @Eloston's rationale [here](https://github.com/Eloston/ungoogled-chromium/issues/447#issuecomment-424150238) on why he further wants to have an option to not use IPv6)

**Comment by rht** [comment]
I grepped throughout the `brave/brave-core` and `brave/brave-browser` codebase (`rg '0x20, '`, `rg 2001:4860`, `rg 2001:7fd`), and wasn't able to find any relevant line corresponding to a DNS server corresponding to Google's or RIPE NCC's. Unless this is being changed elsewhere. The address is hardcoded in chromium's source as far as I can tell.

**Comment by Eloston** [comment]
@tomlowenthal That is because this patch is applied first: https://github.com/Eloston/ungoogled-chromium/blob/master/patches/inox-patchset/0009-disable-google-ipv6-probes.patch

**Comment by diracdeltas** [comment]
@Eloston thanks for the additional detail, that makes sense

@rht i am happy to try to lift https://github.com/Eloston/ungoogled-chromium/blob/master/patches/inox-patchset/0009-disable-google-ipv6-probes.patch to brave, which seems more appropriate than disabling probing altogether. do you have a suggested test case to check (1) probing still works with the change and (2) probing is calling out to non-Google services?

**Comment by rht** [comment]
>  do you have a suggested test case to check (1) probing still works with the change and (2) probing is calling out to non-Google services?

For (1), there is a built-in test suite at https://github.com/chromium/chromium/blob/db29a96e938ab285182c7da5c8b4bd4a9b3c3934/net/dns/host_resolver_impl_unittest.cc#L2844-L2875. For (2), based on the fact that the patch has been functioning for a while, and assuming there is only 1 line where the `kIPv6ProbeAddress` is hardcoded, it should be fine.

In https://github.com/brave/brave-core/pull/972, while it says inox-patchset-0009, I took it from https://github.com/Eloston/ungoogled-chromium/blob/master/patches/inox-patchset/0009-disable-google-ipv6-probes.patch since it is more up to date (though I haven't updated it further for https://github.com/Eloston/ungoogled-chromium/commit/34624717c900bd4d383b441f85cbb9ea05e08c1a#diff-5fd0b0067ce265df5b8170abe187b382).
